csrc/scheduler/resize.cpp

```diff
@@ -373,15 +428,55 @@ void ResizeScheduler::schedule(Fusion* fusion, const HeuristicParams* params) {
   // [..., I0/bdimx(BIDx), bdimx(TIDx), vec_factor]
 
   // Propagate the reference to the other tensors. Note that the
-  // update flag is enabled so to workaround the resize propagation
+  // update flag is enabled to workaround the resize propagation
   // issue. This may not work if there's a tensor that is reshaped
   // from the reference tensor, but that should not be the case as the
   // reference is picked by the same routine used for the pointwise
   // scheduler.
-  scheduler_tools::scheduleLoopDomainsLike(
-      fusion->allTvs(),
-      ref_tv->getLoopDomain(),
-      /*update_loop_domain_only=*/true);
+  //
+  // When an ending static repeat is detected and the repeat ID is
+  // moved to the outermost position, propagation is done separately
+  // between the tensors before the repeat and after the repeat. The
+  // tensors are first grouped into the pre-repeat group and the
+  // post-repeat group, where only the latter group has the repeat
+  // IDs. When propagating the loop domain of the reference tensor,
+  // which has the repeat ID, the full loop domain is propagated only
+  // to the post-repeat group. For the pre-repeat group, the repeat ID
```

Comment:
does it matter if we have `broadcast_out` in `post_repeat` group? Since the `broadcast_out` doesn't have the expanded IDs, I figure whether we propagate the repeated_loop to that target doesn't make a difference in the code...

sorry I'm nitpicking on this, it just bothers me to have the `broadcast_out` as part of the `repeat_tvs` :)

Reply:
As I mentioned above, it can be in either group. Generated code would look different but I believe there's no actual impact, like register usage or loop iteration counts.